City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-448: Breakin change flag and improvements

# [AU-448](https://helsinkisolutionoffice.atlassian.net/browse/AU-448)

## What was done


* Added breaking change flag for webform 3rd party setting. By enabling this, it will deny any editing of submission which are made in older version (Example you could technically edit an application which is in RECEIVED state,  even if there is a new version). This should be used when there is some breaking changes with Avus2 & new version, which will render the old application sending broken.

* Fixed status message printing wrong id while changing status from "In development" -> "Released"

* Locked versioning settings to super user.

* Changed descriptions.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout feature/AU-448-webform-status-improvements`
  * `make fresh`
* Run `make drush-cr`

## How to test


* [ ] Send any application through integration to 'RECEIVED' status.
* [ ] Create new webform version of the given application type. Set breaking change flag on and release the new version. (Also check that the status messages have now correct ids displayed)
* [ ] Try to edit the submission you created on first step. You should not be able to edit it.

Admin stuff:
*  [ ] Login as non user id 1 and check that you cannot change any "webform versioning" settings. Check description that they make sense. 

## Designers review




## Automatic- / Regression tests



## Other PRs


* Link to other PR


[AU-448]: https://helsinkisolutionoffice.atlassian.net/browse/AU-448?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
